gnd/src/main/java/com/google/android/gnd/ui/offlinebasemap/viewer/OfflineBaseMapViewerViewModel.java

```diff
@@ -42,46 +48,64 @@
 public class OfflineBaseMapViewerViewModel extends AbstractViewModel {
 
   @Hot(replays = true)
-  private final FlowableProcessor<OfflineBaseMapViewerFragmentArgs> argsProcessor =
-      BehaviorProcessor.create();
+  private final SingleSubject<OfflineBaseMapViewerFragmentArgs> fragmentArgs =
+      SingleSubject.create();
+
+  @Hot private final PublishProcessor<Nil> removeAreaClicks = PublishProcessor.create();
 
   private final OfflineBaseMapRepository offlineBaseMapRepository;
   private final WeakReference<Context> context;
   public LiveData<Double> areaStorageSize;
   public LiveData<String> areaName;
-  private LiveData<OfflineBaseMap> offlineArea;
+  private final LiveData<OfflineBaseMap> offlineArea;
+  @Inject Navigator navigator;
 
   @Inject
   public OfflineBaseMapViewerViewModel(
-      OfflineBaseMapRepository offlineBaseMapRepository, @ApplicationContext Context context) {
+      OfflineBaseMapRepository offlineBaseMapRepository,
+      @ApplicationContext Context context,
+      Navigator navigator) {
     this.offlineBaseMapRepository = offlineBaseMapRepository;
     this.context = new WeakReference<>(context);
-    this.areaName =
-        LiveDataReactiveStreams.fromPublisher(
-            this.argsProcessor.switchMap(
+    this.navigator = navigator;
+    @Hot
+    // We only need to convert this single to a flowable in order to use it with LiveData.
+    // It still only contains a single offline area returned by getOfflineArea.
+    Flowable<OfflineBaseMap> offlineAreaItemAsFlowable =
+        this.fragmentArgs
```

Comment:
Can we refactor to avoid this deep nesting?

Reply:
done, thanks!